src/common/summarize-profile.js

```diff
@@ -108,23 +94,51 @@ export function summarizeProfile(profile: Profile) {
  * are cached between calls.
  * @returns {function} Function categorizer.
  */
-function functionNameCategorizer() {
-  const cache = new Map();
-  return function functionNameToCategory(name) {
-    const existingCategory = cache.get(name);
-    if (existingCategory !== undefined) {
+function functionNameCategorizer(numFunctions): (number, number[], string[]) => string {
+  const cache = new Array(numFunctions);
+  cache.fill('', 0, numFunctions);
```

Comment:
why do you `fill` the array ? Unless I miss something, I think this is unnecessary and will only take more memory as this will convert a sparse array to a dense array. Maybe @mstange will contradict me here :) but as far as I understand the indices we use in our tables (including the func tables) aren't sequential.

Then of course line 129 you can replace the condition by simply `if (existingCategory)`